Feature/editing

TemVin opened the pull request.

TemVin committed:
add change:
add initial edit input logic

TemVin committed:
add editing

-- added simplified editing book fields

TemVin committed:
add style

--added simple css style

TemVin committed:
add change

-- added appearance / hide edit field and a new component 'Input'

TemVin committed:
add changes

-- added to edit each field with double click and added visual editing

TemVin committed:
add change

-- reduced code size(reduced the number of state variables)

TemVin committed:
add changes

-- added handler for pressing the enter button; part of the code was moved from 'handleBlur' to a new function 'updateInput'

TemVin committed:
add change

-- removed unwanted selector

TemVin committed:
add change

-- added line wrapping

konovalho commented on src/components/Book/Book.jsx:
```diff
@@ -2,27 +2,110 @@ import React, { PureComponent } from 'react';
+    isEditingTitle: false,
+    isEditingAuthor: false,
+    isEditingDescription: false,
+    title: this.props.title,
```
убрать title, author, description

TemVin replied:
Убрал

konovalho commented on src/components/Book/Book.jsx:
```diff
@@ -2,27 +2,110 @@ import React, { PureComponent } from 'react';
+        <div styleName="bookCard">
+          <div styleName="bookOptions">
+            <span styleName="options" onDoubleClick={this.handleEditingTitle}>
+              {this.state.isEditingTitle ? this.createInput(this.state.title, 'title') : this.props.title}
```
не используй локлаьный state this.createInput(this.props.title, 'title')

TemVin replied:
Исправил

konovalho commented on src/components/Book/Book.jsx:
```diff
@@ -2,27 +2,110 @@ import React, { PureComponent } from 'react';
+  }
+
+  handleChange = ({ target: { name, value } }) => {
+    this.setState({
```
меняй не локальный state а сразу данные в redux

TemVin replied:
Хорошо

konovalho commented on src/components/Book/Book.jsx:
```diff
@@ -2,27 +2,110 @@ import React, { PureComponent } from 'react';
+      name={name}
+      value={value}
+      onChange={this.handleChange}
+      onBlur={this.handleBlur}
```
сразу передавать в updateInput

TemVin replied:
Исправил

konovalho commented on src/components/Book/Book.jsx:
```diff
@@ -2,27 +2,110 @@ import React, { PureComponent } from 'react';
   }
 
-  handleEditingBook = (evt) => {
+  handleEditingTitle = () => {
```
явно напрашивается чтоб сократил

konovalho commented on src/components/Book/Book.scss:
```diff
@@ -1,6 +1,59 @@
+.delete {
+  position: relative;
+  display: inline-block;
+  padding: 0.5em 1em;
```
не использовать em, лучше px

TemVin replied:
исправил

TemVin committed:
fixed code review

-- removed local state, replaced 'handleBlur' by  'updateInput', the data in redux changes immediately in handleChange

TemVin committed:
fixed code review

-- corrected styles(added color designation)
